src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,222 @@
+            priority = 3)
+
+    public void writeToFileNullName() throws IOException {
+        Path fileToDeletePath = Paths.get(pathToFolder + "null.json");
+        Cart cartNull = new Cart(null);
+        testParser.writeToFile(cartNull);
+
+        Assert.assertTrue(Files.deleteIfExists(fileToDeletePath), "File was not created");
+    }
+
+
+    @Test (testName = "File contains correct data from Object", groups = {"JsonParser", "Write_to_file", "Smoke"},
+            priority = 1)
+
+    public void writeToFileContainsCorrectData() throws IOException {
+        addingItemsToCart(testCart);
+        testParser.writeToFile(testCart);
+        Reader reader = new FileReader(pathToFile);
+        Cart createdCart = gson.fromJson(reader, Cart.class);
+        reader.close();
+
+        SoftAssert softAssert = new SoftAssert();
+
+        softAssert.assertEquals(createdCart.getCartName(), testCart.getCartName(), "Incorrect Cart Name");
+        softAssert.assertEquals(createdCart.getTotalPrice(), testCart.getTotalPrice(), "Incorrect total price");
+        softAssert.assertAll();
+    }
+
+
+    //Disabled Test
+    @Test (testName = "Exception because file is read only", groups = {"JsonParser", "Write_to_file", "Critical Path"},
+            priority = 2,
+            expectedExceptions = IOException.class)
+    @Ignore
+
+    public void writeToFileReadOnly() throws IOException {
+     newFile.createNewFile();
+     newFile.setReadOnly();
+     System.out.println(newFile.createNewFile());
+    }
+
+
+    //Disabled Test another decision
+    @Test (enabled = false,
+            testName = "Exception because file is read only", groups = {"JsonParser", "Write_to_file", "Critical Path"},
+            priority = 2,
+            expectedExceptions = IOException.class)
+
+    public void writeToFileReadOnly2() throws IOException {
+        newFile.createNewFile();
+        newFile.setReadOnly();
+    }
+
+
+    @Test (testName = "Positive test, reading data from file", groups = {"JsonParser", "Read_From_File", "Smoke"},
+            priority = 1)
+
+    public void readFromFileSmoke() throws IOException {
+
+        addingItemsToCart(testCart);
+        File testCartFile = new File(pathToFile);
+
+        FileWriter writer = new FileWriter(testCartFile);
+        writer.write(gson.toJson(testCart));
+        writer.close();
+
+        Reader reader = new FileReader(pathToFile);
+        Cart expectedCart = gson.fromJson(reader, Cart.class);
+        reader.close();
+
+        Cart actualCart = testParser.readFromFile(testCartFile);
+
+        SoftAssert softAssert = new SoftAssert();
+
+        softAssert.assertEquals (actualCart.getCartName(), expectedCart.getCartName(), "incorrect cart " +
+                        "name value");
+        softAssert.assertEquals (actualCart.getTotalPrice(), expectedCart.getTotalPrice(), "Incorrect total " +
+                        "Price" );
```

Comment:
I forgot to call assertAll, this test will never fail

Reply:
I forgot =((( Now it was added.